# HG changeset patch
# User Rik <rik@octave.org>
# Date 1351622282 25200
#      Tue Oct 30 11:38:02 2012 -0700
# Node ID cf191298a28718988d5ab991cb26ca70c628f40a
# Parent  38e64d9f0fda72c6ac7425be12523b129d9f54f3
# Parent  3649a6012eaac53dfdee5f30343d8582cc8c59a3
maint: Periodic merge of stable to default.

diff --git a/scripts/plot/legend.m b/scripts/plot/legend.m
--- a/scripts/plot/legend.m
+++ b/scripts/plot/legend.m
@@ -971,17 +971,17 @@ function deletelegend2 (h, d, ca, pos, o
           set (ca(i), "units", units);
         end_unwind_protect
       endif
     endif
   endfor
   set (t1, "deletefcn", "");
   delete (t1);
   for i = 1 : numel (hplots)
-    if (strcmp (get (hplots(i), "type"), "line"))
+    if (ishandle (hplots(i)) && strcmp (get (hplots (i), "type"), "line"))
       dellistener (hplots(i), "color");
       dellistener (hplots(i), "linestyle");
       dellistener (hplots(i), "marker");
       dellistener (hplots(i), "markeredgecolor");
       dellistener (hplots(i), "markerfacecolor");
       dellistener (hplots(i), "markersize");
       dellistener (hplots(i), "displayname");
     endif
